gtda/time_series/embedding.py

```diff
@@ -513,7 +509,211 @@ def resample(self, y, X=None):
         check_is_fitted(self)
         yr = column_or_1d(y)
 
-        yr = np.flip(yr)
-        final_index = -self.time_delay_ * (self.dimension_ - 1)
-        yr = np.flip(yr[:final_index:self.stride])
+        final_index = self.time_delay_ * (self.dimension_ - 1)
+        yr = yr[:final_index - 1:-self.stride][::-1]
         return yr
+
+
+@adapt_fit_transform_docs
+class MultiTakensEmbedding(BaseEstimator, TransformerMixin):
```

Comment:
Why Not `TimeDelayEmbedding`? Because the parameters here are fixed and therefore do not actually ensure Taken's theorem.

Reply:
I don't think that communicates the meaning well. Our "Takens embedding" does not ensure that the hypotheses of Taken's theorem are satisfied anyway, and we everywhere use "Takens embedding" as a synonym for "time delay embedding". I am happy to pick another name but it must at least begin to communicate what the difference in intent is in my opinion.

Reply:
You're right, indeed, my suggestion does not makes sense. I will try to think about a new one.

Reply:
A radical proposal is this (but I'm not *seriously* pushing for it):
- What do we mean by "Takens embedding" of a time series? Is it a map from a time series (1D or many-D for multivariate) to a point cloud (always 2D)?
- If "yes" above, then just as "VietorisRipsPersistence" computes the VR-persistence of each entry in a collection, and returns the stacking of all resulting diagrams, the `TakensEmbedding` class should compute the Takens embedding of each time series in a collection, and return a collection of point clouds.

So the radical point of view is that the *new* transformer should be the one called `TakensEmbedding`, while the old one should be renamed to something like `SingleTakensEmbedding`.

Just adding this to the fire because I like to watch the world burn.

Reply:
I completely agree with your comment.

Reply:
> A radical proposal is this (but I'm not _seriously_ pushing for it):

Is it a seriously-considered proposal?

Reply:
It seems to be receiving more attention than I had anticipated...